7.application_configuration.md

```diff
@@ -0,0 +1,186 @@
+
+### ScopeRef
+
+The scopeRef section will index a scope instance by `apiVersion`, `kind` and `name`.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `apiVersion` | `string` | Y | | The apiVersion of the Scope. |
+| `kind` | `string` | Y | | The kind of the Scope. |
+| `name` | `string` | Y | | The name of the Scope. |
+
+> The name field is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.
+
+### Trait
+
+The trait section defines an instance of a trait.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `trait` | [`TraitData`](#traitdata) | N | | Supply all information needed by a trait definition. |
+
+### TraitData
+
+The TraitData section contains the instantiation of the [schema](3.traits.md#DefinitionRef) referenced by the corresponding traitDefinition. We can use the schema to validated this section.
+
+It's __RECOMMENDED__ that `apiVersion` and `kind` are contained like below:
+
+``` yaml
+- trait:
+    apiVersion: core.oam.dev/v1alpha2
+    kind: ManualScalerTrait
+    spec:
+      replica: 2
+```
+
+### ParameterValue
+
+Values supplied to parameters that are used to override the parameters exposed by other types.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `name` | `string` | Y | | The name of the component parameter to provide a value for. |
+| `value` | `string` | Y || The value of this parameter. |
+
+A `name` is any Unicode letter or numeric character, `_`, `-`, and `.`. A `value` is any sequence of printable Unicode characters.
+
+## Example
+
+The following is an example of a complete YAML file that expresses configuration and traits. This example illustrates the four configurational elements above.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: my-app-deployment
+  annotations:
+    version: v1.0.0
+    description: "Description of this deployment"
+spec:
+  components:
+    - componentName: my-web-app-component
+      parameterValues:
+        - name: PARAMETER_NAME
+          value: SUPPLIED_VALUE
+        - name: ANOTHER_PARAMETER
+          value: "AnotherValue"
+      traits:
+        - name: manualscaler.core.oam.dev
+          version: v1
+          spec:
+            replicaCount: 3
+      scopes:
+        - scopeRef:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: NetworkScope
+            name: example-vpc-network
+```
+
+The example above illustrates a complete application configuration scopes, component instances, their traits, and parameter overrides that allow users to specify values at deployment time from the command line.
```

Comment:
- The example above illustrates a complete application configuration scopes, component instances, their trait, and ...
+ The example above illustrates a complete application configuration, containing scopes, component instances with corresponding traits, and ...